WordPress/Classes/Networking/ReaderPostServiceRemote.m

```diff
@@ -266,45 +287,121 @@ - (RemoteReaderPost *)formatPostDictionary:(NSDictionary *)dict
+
+    NSDictionary *tags = [self primaryAndSecondaryTagsFromPostDictionary:dict];
+    if (tags) {
+        post.primaryTag = [tags stringForKey:TagKeyPrimary];
+        post.primaryTagSlug = [tags stringForKey:TagKeyPrimarySlug];
+        post.secondaryTag = [tags stringForKey:TagKeySecondary];
+        post.secondaryTagSlug = [tags stringForKey:TagKeySecondarySlug];
+    }
+
+    post.isExternal = [[dict numberForKey:PostRESTKeyIsExternal] boolValue];
+    post.isJetpack = [[dict numberForKey:PostRESTKeyIsJetpack] boolValue];
+    post.wordCount = [dict numberForKey:PostRESTKeyWordCount];
+    post.readingTime = [self readingTimeForWordCount:post.wordCount];
+
     if ([dict arrayForKeyPath:@"discover_metadata.discover_fp_post_formats"]) {
-        post.sourceAttribution = [self sourceAttributionFromDictionary:[dict dictionaryForKey:@"discover_metadata"]];
+        post.sourceAttribution = [self sourceAttributionFromDictionary:[dict dictionaryForKey:PostRESTKeyDiscoverMetadata]];
     }
     return post;
 }
 
+- (NSDictionary *)primaryAndSecondaryTagsFromPostDictionary:(NSDictionary *)dict
+{
+    NSString *primaryTag = @"";
+    NSString *primaryTagSlug = @"";
+    NSString *secondaryTag = @"";
+    NSString *secondaryTagSlug = @"";
+    NSString *editorialTag;
+    NSString *editorialSlug;
+
+    NSArray *remoteTags = [[dict dictionaryForKey:PostRESTKeyTags] allValues];
+    if (remoteTags) {
+        NSInteger postCount = 0;
+
+        for (NSDictionary *tag in remoteTags) {
+            NSInteger count = [[tag numberForKey:PostRESTKeyPostCount] integerValue];
+            if (count > postCount) {
+                secondaryTag = primaryTag;
+                secondaryTagSlug = primaryTagSlug;
+
+                primaryTag = [tag stringForKey:PostRESTKeyName] ?: @"";
+                primaryTagSlug = [tag stringForKey:PostRESTKeySlug] ?: @"";
+
+                postCount = count;
+            }
+        }
+    }
+
+    NSDictionary *editorial = [dict dictionaryForKey:PostRESTKeyEditorial];
+    if (editorial) {
+        editorialSlug = [editorial stringForKey:PostRESTKeyHighlightTopic];
+        editorialTag = [editorial stringForKey:PostRESTKeyHighlightTopicTitle] ?: [editorialSlug capitalizedString];
+    }
+
+    if (editorialSlug) {
+        secondaryTag = primaryTag;
+        secondaryTagSlug = primaryTagSlug;
+        primaryTag = editorialTag;
+        primaryTagSlug = editorialSlug;
+    }
+
+    return @{
+             TagKeyPrimary:primaryTag,
+             TagKeyPrimarySlug:primaryTagSlug,
+             TagKeySecondary:secondaryTag,
+             TagKeySecondarySlug:secondaryTagSlug,
+             };
+}
+
+- (NSNumber *)readingTimeForWordCount:(NSNumber *)wordCount
+{
+    NSInteger avgWordsPerMinuteRead = 250;
+    NSInteger minimumMinutesToRead = 2;
+    NSInteger count = [wordCount integerValue];
+
+    NSInteger minutesToRead = count / avgWordsPerMinuteRead;
+    if (minutesToRead < minimumMinutesToRead) {
+        return @(0);
```

Comment:
It should be zero, but I think I could find a better name for `minimumMinuteToRead`. I think the name makes it confusing why we'd want a zero returned.
They should totally be constants.  I'll make that change :)